raster/src/main/scala/geotrellis/raster/costdistance/SimpleCostDistance.scala

```diff
@@ -0,0 +1,204 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster.costdistance
+
+import geotrellis.raster._
+
+import java.util.PriorityQueue
+
+
+/**
+  * Object housing various functions related to Cost-Distance
+  * computations.
+  */
+object SimpleCostDistance {
+
+  type Cost = (Int, Int, Double, Double) // column, row, friction, cost
+  type Q = PriorityQueue[Cost]
+  type EdgeCallback = (Cost => Unit)
+
+  /**
+    * NOP EdgeCallback
+    */
+  def nop(cost: Cost): Unit = {}
+
+  /**
+    * Generate a Queue suitable for working with a tile of the given
+    * dimensions.
+    *
+    * @param  cols  The number of columns of the friction tile
+    * @param  rows  The number of rows of the friction tile
+    */
+  def generateEmptyQueue(cols: Int, rows: Int): Q = {
+    new PriorityQueue(
+      (cols*16 + rows*16), new java.util.Comparator[Cost] {
+        override def equals(a: Any) = a.equals(this)
+        def compare(a: Cost, b: Cost) = a._4.compareTo(b._4)
+      })
+  }
+
+  /**
+    * Generate an empty double-valued array tile of the correct
+    * dimensions.
+    *
+    * @param  cols  The number of cols of the friction tile (and therefore the cost tile)
+    * @param  rows  The number of rows of the frition tile and cost tiles
+    */
+  def generateEmptyCostTile(cols: Int, rows: Int): DoubleArrayTile =
+    DoubleArrayTile.empty(cols, rows)
+
+  /**
+    * Generate a cost-distance raster based on a set of starting
+    * points and a friction raster.  This is an implementation of the
+    * standard algorithm cited in the "previous work" section of [1].
+    *
+    * 1. Tomlin, Dana.
+    *    "Propagating radial waves of travel cost in a grid."
+    *    International Journal of Geographical Information Science 24.9 (2010): 1391-1413.
+    *
+    * @param  frictionTile  Friction tile; pixels are interpreted as "second per meter"
+    * @param  points        List of starting points as tuples
+    * @param  maxCost       The maximum cost before pruning a path (in units of "seconds")
+    * @param  resolution    The resolution of the tiles (in units of "meters per pixel")
+    */
+  def apply(
+    frictionTile: Tile,
+    points: Seq[(Int, Int)],
+    maxCost: Double = Double.PositiveInfinity,
+    resolution: Double = 1
+  ): DoubleArrayTile = {
+    val cols = frictionTile.cols
+    val rows = frictionTile.rows
+    val costTile = generateEmptyCostTile(cols, rows)
+    val q: Q = generateEmptyQueue(cols, rows)
+
+    var i = 0; while (i < points.length) {
```

Comment:
Why don't to use just `cfor`? It's a common dep for the whole raster package. Just less ugly ^^'.